packages/job-manager/src/batch/batch.ts

```diff
@@ -26,6 +28,61 @@ export class Batch {
         @inject(Logger) private readonly logger: Logger,
     ) {}
 
+    public async getPoolTaskProcessingRatio(): Promise<number> {
+        const resultVariableName = 'taskProcessingRatioResult';
+        const poolTaskProcessingRatioFormula = `
+        // The script will calculate the tasks processing ratio returned in $taskProcessingRatio variable as
+        // double value and will return -1 when there are no active tasks or sampling data is insufficient
+
+        // the time interval for data sampling
+        $sampleTimeIntervalMinutes = 15 * TimeInterval_Minute;
+        $samplePercentThreshold = 70;
+
+        // get sample data for pending tasks (the sum of active and running tasks)
+        $pendingTasksSamplePercent = $PendingTasks.GetSamplePercent($sampleTimeIntervalMinutes);
+        $pendingTasksVector = $PendingTasks.GetSample(TimeInterval_Minute, $sampleTimeIntervalMinutes);
+        $pendingTasksAvg = avg($pendingTasksVector);
+
+        // get sample data for running tasks
+        $runningTasksSamplePercent = $RunningTasks.GetSamplePercent($sampleTimeIntervalMinutes);
+        $runningTasksVector = $RunningTasks.GetSample(TimeInterval_Minute, $sampleTimeIntervalMinutes);
+        $runningTasksAvg = avg($runningTasksVector);
+
+        // calculate the tasks processing ratio
+        $taskProcessingRatio = $runningTasksAvg / ($pendingTasksAvg > 0 ? $pendingTasksAvg : 1);
```

Comment:
should this be?
```suggestion
        $taskProcessingRatio = ($runningTasksAvg > 0 || $pendingTasksAvg > 0) ? $runningTasksAvg / ($pendingTasksAvg +  $runningTasksAvg) : 0;
```

Reply:
taskProcessingRatio is percent ratio of running tasks to all tasks in progress (that is sum of active and running tasks). This line make sure that there is no division on 0. Next line will return  -1 when no tasks been processed ie pendingTaskCount = 0

---
In reply to: [305944851](https://github.com/microsoft/accessibility-insights-service/pull/221#discussion_r305944851) [](ancestors = 305944851)